transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -842,10 +843,8 @@ public ChannelFuture writeAndFlush(Object msg) {
     }
 
     private static void notifyOutboundHandlerException(Throwable cause, ChannelPromise promise) {
-        if (!promise.tryFailure(cause) && !(promise instanceof VoidChannelPromise)) {
-            if (logger.isWarnEnabled()) {
-                logger.warn("Failed to fail the promise because it's done already: {}", promise, cause);
-            }
+        if (!(promise instanceof VoidChannelPromise)) {
```

Comment:
@Scottmitch actual we can not even use `isVoid()` in 4.1 as it is not part of the `Future` interface but only of `ChannelFuture` :(